### PR TITLE
Add exporters for GDPR

### PR DESCRIPTION
**Pull request by dharmin**
Activity with attachments Exporter
Comments with attachments Exporter
Media Views Exporter
Media Likes Exporter

### CODE REVIEW

File: app/main/controllers/template/rtmedia-filters.php [hunk 1]
```diff
@@ -853,3 +853,68 @@ function rtmedia_gallery_shortcode_parameter_pre_filter_callback( $attr ) {
...
+
+/**
+ * Add exporters to queue
+ *
+ * @param  array $exporters Exporter queue.
+ * @return array
+ */
+function register_rtmedia_exporter( $exporters ) {
+	$exporters['buddypress-media-shortcode-uploads'] = array(
+		'exporter_friendly_name' => __( 'rtMedia Shortcode Uploads' ),
```
Comment (milindmore22) [inline app/main/controllers/template/rtmedia-filters.php]: `esc_html__` is preferred and missing text domain `buddypress-media`

File: app/main/controllers/template/rtmedia-functions.php [hunk 2]
```diff
@@ -4179,3 +4179,562 @@ function rtmedia_number_to_human_readable( $n ) {
...
+	global $wpdb;
+
+	$query = $wpdb->prepare(
+		'SELECT * FROM ' . $wpdb->prefix . "bp_activity WHERE user_id=%d and type='rtmedia_update'  LIMIT %d OFFSET %d",
+		$user_data->ID,
+		$number,
+		$number * ( $page - 1 )
+	);
+
+	$activities = $wpdb->get_results( $query );
```
Comment (milindmore22) [inline app/main/controllers/template/rtmedia-functions.php]: use object cache for resultset

File: app/main/controllers/template/rtmedia-functions.php [hunk 3]
```diff
@@ -4179,3 +4179,562 @@ function rtmedia_number_to_human_readable( $n ) {
...
+
+		$activity_content = wp_strip_all_tags( $activity->content );
+		$activity_date    = $activity->date_recorded;
+		$activity_id      = $activity->id;
+		$item_id          = 'activity-' . $activity_id;
+		$group_id         = 'activity';
+		$group_label      = __( 'rtMedia Activities' );
+
+		$query          = $wpdb->prepare( 'SELECT media_id, media_title FROM ' . $wpdb->prefix . 'rt_rtm_media WHERE activity_id=%d', $activity_id );
+		$results        = $wpdb->get_results( $query );
```
Comment (milindmore22) [inline app/main/controllers/template/rtmedia-functions.php]: use object cache for resultset

File: app/main/controllers/template/rtmedia-functions.php [hunk 4]
```diff
@@ -4179,3 +4179,562 @@ function rtmedia_number_to_human_readable( $n ) {
...
+	$activities = $wpdb->get_results( $query );
+
+	foreach ( $activities as $activity ) {
+
+		$activity_content = wp_strip_all_tags( $activity->content );
+		$activity_date    = $activity->date_recorded;
+		$activity_id      = $activity->id;
+		$item_id          = 'activity-' . $activity_id;
+		$group_id         = 'activity';
+		$group_label      = __( 'rtMedia Activities' );
```
Comment (milindmore22) [inline app/main/controllers/template/rtmedia-functions.php]: esc_html__ and text domain

File: app/main/controllers/template/rtmedia-functions.php [hunk 5]
```diff
@@ -4179,3 +4179,562 @@ function rtmedia_number_to_human_readable( $n ) {
...
+
+		$query          = $wpdb->prepare( 'SELECT media_id, media_title FROM ' . $wpdb->prefix . 'rt_rtm_media WHERE activity_id=%d', $activity_id );
+		$results        = $wpdb->get_results( $query );
+		$activity_count = $wpdb->num_rows;
+		$attachments    = '';
+
+		foreach ( $results as $result ) {
+			$url          = wp_get_attachment_url( $result->media_id );
+			$attachments .= $result->media_title . " : <a href='$url'>$url</a><br />";
```
Comment (milindmore22) [inline app/main/controllers/template/rtmedia-functions.php]: use esc_url

File: app/main/controllers/template/rtmedia-functions.php [hunk 6]
```diff
@@ -4179,3 +4179,562 @@ function rtmedia_number_to_human_readable( $n ) {
...
+				'name'  => __( 'Activity Date' ),
+				'value' => $activity_date,
+			),
+			array(
+				'name'  => __( 'Activity Content' ),
+				'value' => $activity_content,
+			),
+			array(
+				'name'  => __( 'Attachments' ),
+				'value' => empty( $attachments ) ? 'No attachments' : $attachments,
```
Comment (milindmore22) [inline app/main/controllers/template/rtmedia-functions.php]: needs translation and escaping

File: app/main/controllers/template/rtmedia-functions.php [hunk 7]
```diff
@@ -4179,3 +4179,562 @@ function rtmedia_number_to_human_readable( $n ) {
...
+			'data' => $export_items,
+			'done' => true,
+		);
+	}
+
+	global $wpdb;
+
+	$query = $wpdb->prepare( 'SELECT media.media_id, media.media_title, media.upload_date, album.media_title AS album_title FROM ' . $wpdb->prefix . 'rt_rtm_media AS media, ' . $wpdb->prefix . 'rt_rtm_media AS album WHERE media.album_id=album.id AND media.activity_id=0 AND media.media_author=%d LIMIT %d OFFSET %d', $user_data->ID, $number, $number * ( $page - 1 ) );
+
+	$media       = $wpdb->get_results( $query );
```
Comment (milindmore22) [inline app/main/controllers/template/rtmedia-functions.php]: needs caching

File: app/main/controllers/template/rtmedia-functions.php [hunk 8]
```diff
@@ -4179,3 +4179,562 @@ function rtmedia_number_to_human_readable( $n ) {
...
 
+/**
+ * Activity exporter with attachments for GDPR
+ *
+ * @param  string $email_address user email address.
+ * @param  int    $page          page no to fetch data from.
+ *
+ * @return array
+ */
+function rtmedia_activity_exporter( $email_address, $page = 1 ) {
```
[the remark this answers is not in the document]
Comment (dharmin) [reply app/main/controllers/template/rtmedia-functions.php]: yes, it collects data page wise to avoid timeout.

**Pull request merged**